### PR TITLE
Fix directory structure to allow for programmatic usage of the V2+ version

### PR DESCRIPTION
**Pull request by christophetd**
## Background

Stratus Red Team is generally used as a CLI but also supports [programmatic usage](https://stratus-red-team.cloud/user-guide/programmatic-usage/), for instance:

```go
package main

import (
	"fmt"
	"github.com/datadog/stratus-red-team/pkg/stratus"
	_ "github.com/datadog/stratus-red-team/pkg/stratus/loader" // Note: This import is needed
	stratusrunner "github.com/datadog/stratus-red-team/pkg/stratus/runner"
)

func main() {
	ttp := stratus.GetRegistry().GetAttackTechniqueByName("aws.defense-evasion.cloudtrail-stop")
	stratusRunner := stratusrunner.NewRunner(ttp, stratusrunner.StratusRunnerNoForce)

	stratusRunner.WarmUp()
	defer stratusRunner.CleanUp()
	stratusRunner.Detonate()
}
```

On June 3rd 2022, we bumped the Stratus Red Team version from 1.8.0 to 2.0.0 because of a breaking change. Unexpectedly, this caused any Stratus Red Team programmatic usage to fail with the following error message:

```
$ cat go.mod
module my-module
go 1.18
require github.com/datadog/stratus-red-team v2.0.0

$ go mod tidy
go: errors parsing go.mod:
/tmp/my-project/go.mod:3: require github.com/datadog/stratus-red-team: version "v2.0.0" invalid: should be v0 or v1, not v2
```

After some investigation, it appears that bumping to 2.0.0 might have been a bad decision. Go actively dislikes major version bumps, and requires you to **move all your code on a different module**. See https://golang.cafe/blog/how-to-upgrade-to-a-major-version-in-go.html for more detail about this

## This PR

* Move all the code to a different module (`github.com/datadog/stratus-red-team` => `github.com/datadog/stratus-red-team/v2`)
* Fix imports, examples and build scripts to use this new path for imports

## New usage

Install with

```
$ go get github.com/datadog/stratus-red-team/v2@294f96a
# This will be go get github.com/datadog/stratus-red-team/v2@v2.2.0 when this PR is merged
```

Sample code:

```go
package main

import (
	"fmt"
	"github.com/datadog/stratus-red-team/v2/pkg/stratus" // <= New import path
	_ "github.com/datadog/stratus-red-team/v2/pkg/stratus/loader" // <= New import path
	stratusrunner "github.com/datadog/stratus-red-team/v2/pkg/stratus/runner" // <= New import path
)

func main() {
	ttp := stratus.GetRegistry().GetAttackTechniqueByName("aws.defense-evasion.cloudtrail-stop")
	stratusRunner := stratusrunner.NewRunner(ttp, stratusrunner.StratusRunnerNoForce)

	stratusRunner.WarmUp()
	defer stratusRunner.CleanUp()
	stratusRunner.Detonate()
}
```

### CODE REVIEW

File: docs/user-guide/programmatic-usage.md [hunk 1]
```diff
@@ -24,4 +24,4 @@ See https://github.com/DataDog/stratus-red-team/tree/main/examples
 
 ## Reference
 
-https://pkg.go.dev/github.com/datadog/stratus-red-team/pkg/stratus
+https://pkg.go.dev/github.com/datadog/stratus-red-team/v2/pkgstratus
```
Comment (xen0ldog) [inline docs/user-guide/programmatic-usage.md:27]: This should be `https://pkg.go.dev/github.com/datadog/stratus-red-team/v2/pkg/stratus`. `https://pkg.go.dev/github.com/datadog/stratus-red-team/v2/pkgstratus` returns 404

File: v2/internal/providers/kubernetes.go [hunk 2]
```diff
@@ -2,7 +2,7 @@ package providers
 
 import (
 	"context"
-	"github.com/datadog/stratus-red-team/internal/utils"
+	"github.com/datadog/stratus-red-team/v2/internal/utils"
```
Comment (xen0ldog) [inline v2/internal/providers/kubernetes.go:5]: Just to consistent with other use, shouldn't we call this `utils2`?

```suggestion
         utils2 "github.com/datadog/stratus-red-team/v2/internal/utils"
```

Comment (xen0ldog) [reply v2/internal/providers/kubernetes.go:5]: There is a use case on lines https://github.com/DataDog/stratus-red-team/pull/146/files#diff-8106b62550d342b69bc5acbffc4431b731809ded0e93090f12ae7b99092b79b1L58-R60

Comment (christophetd) [reply v2/internal/providers/kubernetes.go:5]: fixed in [b38133c](https://github.com/DataDog/stratus-red-team/pull/146/commits/b38133c9bbeba4ae777a49daec5c79d957abf752)

**Pull request merged**